Lib/test/test_pathlib.py

```diff
@@ -345,6 +345,8 @@ def test_match_common(self):
         self.assertTrue(P('/a/b.py').match('/a/*.py'))
         self.assertFalse(P('/ab.py').match('/a/*.py'))
         self.assertFalse(P('/a/b/c.py').match('/a/*.py'))
+        # Multi-part glob-style pattern
+        self.assertFalse(P('/a/b/c.py').match('/**/*.py'))
```

Comment:
Thanks for the PR. Is it worth adding `self.assertTrue(P('/a/b/c.py').match('/a/**/*.py'))` where 
* `c.py` matches with `*.py`
* `b` matches with `**`
* `a` matches with `a`

Internally the function first checks for the length of the split parts to be equal along with transforming '**' as below that matches with 'b' : 

```python
>>> import fnmatch
>>> fnmatch.translate('**')
'(?s:.*.*)\\Z'
```

I don't know if this can be documented or we can just leave it without mentioning the support of this since it requires documenting the internal implementation to say how this match occurs. It's just a suggestion though. 

I will wait for review from @serhiy-storchaka with respect to https://bugs.python.org/msg285311

Reply:
More tests don't hurt when they aren't overly restrictive and this doesn't strike me as such.

Reply:
committed in https://github.com/python/cpython/pull/11171/commits/9688d5385fa31bb2aa7dc59a730c73e3f45584e3